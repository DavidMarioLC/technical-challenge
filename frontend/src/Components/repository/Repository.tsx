import "./repository.css";
import Button from '../common/button/Button'
import Tag from '../common/tag/Tag'
import LinkDetails  from "./linkDetails/LinkDetails";
import {TiStarOutline} from 'react-icons/ti'
import {GoPrimitiveDot} from 'react-icons/go'
import {AiOutlineStar} from 'react-icons/ai'
import {FaBalanceScale} from 'react-icons/fa';
import { IOwner,Ilicense } from "../../interfaces/repository.interface";
import {formatDate} from '../../utils/formatDate'
import {formatStars} from '../../utils/formatStars';


interface Props {
  name:string;
  description:string;
  language:string;
  visibility:string;
  owner:IOwner;
  pushed_at:Date;
  license?:Ilicense;
  stargazers_count:number
}


const Repository = ({name,description,language,visibility,owner,pushed_at,license,stargazers_count}:Props) => {

  return (
    <article className="repository">
              <div className="repository__content">
               <div className="repository__header">
                 <a className="repository__name" href={`${owner.html_url}/${name}`} target="_blank" rel="noreferrer">{name}</a>
                 <span className="repository__tag">{visibility}</span>
                </div>
                
                <div className="repository__description"><p>{description}</p></div>
                <div className="repository__tags">
                  <Tag value='css'/>
                  <Tag value='html'/>
                  <Tag value='javascript'/>
                  <Tag value='reactjs'/>
                  <Tag value='sass'/>
                </div>
                <div className="repository__details">
                  {language&&<LinkDetails icon={GoPrimitiveDot} type='language' value={language} />}
                 {stargazers_count>0&&<LinkDetails icon={AiOutlineStar} type='star' value={formatStars(stargazers_count)} />}
                  {license&&<LinkDetails icon={FaBalanceScale} type='license' value={license?.name} />}
                  <LinkDetails  type='date' value={`Updated ${formatDate(pushed_at)}` } />
                </div>
              </div>  

              <div className="repository__actions">
                <Button size="small" icon={TiStarOutline} value="Star"/>
              </div>
              
          </article>
  )
}

export default Repository
