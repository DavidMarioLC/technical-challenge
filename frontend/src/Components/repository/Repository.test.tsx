import renderer from 'react-test-renderer';
import Repository from './Repository';


it('render Repository correctly', () => {
  const tree = renderer.create(<Repository 
    name={'name repository'} 
    description={'description repository'}
    language={'language'}
    visibility={'public'}
    owner={{
      avatar_url: '',
      events_url:'',
      followers_url: '',
      following_url: '',
      gists_url: '',
      gravatar_id: '',
      html_url: '',
      id: 0,
      login: '',
      node_id: '',
      organizations_url: '',
      received_events_url: '',
      repos_url: '',
      site_admin: true,
      starred_url: '',
      subscriptions_url: '',
      type: '',
      url: ''
    }}

    pushed_at={new Date()}
    license={{key: '',
      name:'', 
      spdx_id: '', 
      url: '', 
      node_id: '', }}
    stargazers_count={12}
    
    />).toJSON();
  expect(tree).toMatchSnapshot();
});