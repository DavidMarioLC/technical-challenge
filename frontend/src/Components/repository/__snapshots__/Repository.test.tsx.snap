// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Repository correctly 1`] = `
<article
  className="repository"
>
  <div
    className="repository__content"
  >
    <div
      className="repository__header"
    >
      <a
        className="repository__name"
        href="/name repository"
        rel="noreferrer"
        target="_blank"
      >
        name repository
      </a>
      <span
        className="repository__tag"
      >
        public
      </span>
    </div>
    <div
      className="repository__description"
    >
      <p>
        description repository
      </p>
    </div>
    <div
      className="repository__tags"
    >
      <a
        className="tag"
        href="https://github.com/topics/css"
        rel="noreferrer"
        target="_blank"
      >
        css
      </a>
      <a
        className="tag"
        href="https://github.com/topics/html"
        rel="noreferrer"
        target="_blank"
      >
        html
      </a>
      <a
        className="tag"
        href="https://github.com/topics/javascript"
        rel="noreferrer"
        target="_blank"
      >
        javascript
      </a>
      <a
        className="tag"
        href="https://github.com/topics/reactjs"
        rel="noreferrer"
        target="_blank"
      >
        reactjs
      </a>
      <a
        className="tag"
        href="https://github.com/topics/sass"
        rel="noreferrer"
        target="_blank"
      >
        sass
      </a>
    </div>
    <div
      className="repository__details"
    >
      <div
        className="linkDetails"
      >
        <svg
          className="dot dot--language"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 8 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="linkDetail-text"
        >
          language
        </span>
      </div>
      <div
        className="linkDetails"
      >
        <svg
          className="dot dot--12"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
          />
        </svg>
        <span
          className="linkDetail-text"
        >
          12
        </span>
      </div>
      <div
        className="linkDetails"
      >
        <svg
          className="dot dot--"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 336h-.02c0-16.18 1.34-8.73-85.05-181.51-17.65-35.29-68.19-35.36-85.87 0C-2.06 328.75.02 320.33.02 336H0c0 44.18 57.31 80 128 80s128-35.82 128-80zM128 176l72 144H56l72-144zm511.98 160c0-16.18 1.34-8.73-85.05-181.51-17.65-35.29-68.19-35.36-85.87 0-87.12 174.26-85.04 165.84-85.04 181.51H384c0 44.18 57.31 80 128 80s128-35.82 128-80h-.02zM440 320l72-144 72 144H440zm88 128H352V153.25c23.51-10.29 41.16-31.48 46.39-57.25H528c8.84 0 16-7.16 16-16V48c0-8.84-7.16-16-16-16H383.64C369.04 12.68 346.09 0 320 0s-49.04 12.68-63.64 32H112c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h129.61c5.23 25.76 22.87 46.96 46.39 57.25V448H112c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <span
          className="linkDetail-text"
        >
          
        </span>
      </div>
      <div
        className="linkDetails"
      >
        <span
          className="linkDetail-text"
        >
          Updated a few seconds ago
        </span>
      </div>
    </div>
  </div>
  <div
    className="repository__actions"
  >
    <button
      className="btn btn--small undefined"
    >
      <svg
        baseProfile="tiny"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.855 20.966c-.224 0-.443-.05-.646-.146l-.104-.051-4.107-2.343-4.107 2.344-.106.053c-.488.228-1.085.174-1.521-.143-.469-.34-.701-.933-.586-1.509l.957-4.642-1.602-1.457-1.895-1.725-.078-.082c-.375-.396-.509-.97-.34-1.492.173-.524.62-.912 1.16-1.009l.102-.018 4.701-.521 1.946-4.31.06-.11c.262-.473.764-.771 1.309-.771.543 0 1.044.298 1.309.77l.06.112 1.948 4.312 4.701.521.104.017c.539.1.986.486 1.158 1.012.17.521.035 1.098-.34 1.494l-.078.078-3.498 3.184.957 4.632c.113.587-.118 1.178-.59 1.519-.252.182-.556.281-.874.281zm-8.149-6.564c-.039.182-.466 2.246-.845 4.082l3.643-2.077c.307-.175.684-.175.99 0l3.643 2.075-.849-4.104c-.071-.346.045-.705.308-.942l3.1-2.822-4.168-.461c-.351-.039-.654-.26-.801-.584l-1.728-3.821-1.726 3.821c-.146.322-.45.543-.801.584l-4.168.461 3.1 2.822c.272.246.384.617.302.966z"
        />
      </svg>
      Star
    </button>
  </div>
</article>
`;
